include/seqan3/core/type_list.hpp

```diff
@@ -31,10 +35,485 @@ using type_list = meta::list<types...>;
+//!\endcond
+using front = typename decltype(detail::front_impl(list_t{}))::type;
+
+/*!\brief Return the last type from the type list.
+ * \tparam list_t The type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <seqan3::detail::TypeList list_t>
+//!\cond
+    requires (size<list_t> > 0)
+//!\endcond
+using back = typename decltype(detail::back_impl(list_t{}))::type;
+
+/*!\brief Join two seqan3::type_list s into one.
+ * \tparam list1_t The first (input) type list.
+ * \tparam list2_t The second (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <seqan3::detail::TypeList list1_t, seqan3::detail::TypeList list2_t>
+using concat = decltype(detail::concat_impl(list1_t{}, list2_t{}));
+
+/*!\brief Return a seqan3::type_list of all the types in the type list, except the first.
+ * \tparam list_t The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <seqan3::detail::TypeList list_t>
+//!\cond
+    requires (size<list_t> > 0)
+//!\endcond
+using drop_front = decltype(detail::drop_front_impl(list_t{}));
+
+/*!\brief Return a seqan3::type_list of the first `n` types in the input type list.
+ * \tparam n        The target size; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using take = typename decltype(detail::split_after_impl<n>(type_list<>{}, list_t{}))::first_type;
+
+/*!\brief Return a seqan3::type_list of the types in the input type list, except the first `n`.
+ * \tparam n        The amount to drop; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using drop = typename decltype(detail::split_after_impl<n>(type_list<>{}, list_t{}))::second_type;
+
+/*!\brief Return a seqan3::type_list of the last `n` types in the input type list.
+ * \tparam n        The target size; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `size<list_t> - n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using take_last = typename decltype(detail::split_after_impl<size<list_t> - n>(type_list<>{}, list_t{}))::second_type;
```

Comment:
First of all this would be the same as:
```suggestion
using take_last = drop<size<list_t> - n>;
```
right?
And I am wondering if this should not be modeled with negative indices`take<-n, list_t>`? 
That would it make consistent with `at`
So:

```c++
template <ptrdiff_t n, seqan3::detail::TypeList list_t>
//!\cond
    requires (n < 0 && -n <= size<list_t>)
//!\endcond
using take = drop<size<list_t> + n>;
```

Reply:
I agree, it can be written shorter. See above for why I am opposed to negative idnexing there. 

(Believe me, I wanted this, I designed it that way and it confused me so much I got convinced it wasn't a good idea)